### ISSUE TITLE
issue with type import()s that resolve to declaration files: "Namespace X has no exported member Y"

### ISSUE THREAD

**Issue by dgoldstein0** [microsoft/TypeScript#41006]
**TypeScript Version:**  every version I've tried - repros from 3.3.3 all the way to nightly

**Search Terms:** namespace dynamic type only import no exported member

**Code**

```ts
import * as x from "tslib";

const assign: import("tslib").__assign = x.__assign; // Namespace '"tslib"' has no exported member '__assign'.(2694)
let module: typeof import("tslib");

const assign2: typeof module.__assign = x.__assign; // works
```

**Expected behavior:** no type errors

**Actual behavior:** the `import("tslib")` line always has an error.

I first ran into this using eventemitter3 instead of tslib, but I figured tslib is a more universal dependency.

**Playground Link:** https://www.typescriptlang.org/play?ts=4.1.0-dev.20201009#code/JYWwDg9gTgLgBAKjgQwM5wB5wGZQiOAIhlQBtgAjQgbgFgAoBgYwgDtV41VgBzVgLjihIsABTEylQgEoAdAH15XXqzgBeTAqWpufOo3qkApvBAQAJgFdjgmAE8wRiNiHhoMcSXJVp+5mw4UHRUAJlsHJxczK2MtZT51TUV41n0gA

**Related Issues:** #28660 looks similar

**Comment by andrewbranch** [comment]
You’re [missing a `typeof`](https://www.typescriptlang.org/play?ts=4.1.0-dev.20201009#code/JYWwDg9gTgLgBAKjgQwM5wB5wGZQiOAIhlQBtgAjQgbgFgAoBgYwgDtV41VgBzVgLjgwAnmACmEbHFCRYACmJlKhAJQA6APoauvVnAC8mTdtTc+dRvVJj4ICABMArtcEjxk6eGgwFJclRULZjYOFFNdACZXUQkpOydrYx0+AyMtZNYLIA).

**Comment by dgoldstein0** [comment]
oh, hm.

I believe I had tried `const assign: (typeof import("tslib")).__assign = x.__assign;` but that gives a syntax error.  It doesn't make much sense to me that without the extra parentheses it works - I'd expect the `._assign` to have higher precedence than the `typeof`.
